In [1]:
import pandas as pd
from sklearn.datasets import load_wine

In [2]:
wine = load_wine()

In [3]:
dir(wine)

['DESCR', 'data', 'feature_names', 'frame', 'target', 'target_names']

In [4]:
df = pd.DataFrame(wine.data, columns = wine.feature_names)

In [5]:
df

alcohol  malic_acid   ash  alcalinity_of_ash  magnesium  total_phenols  \
0      14.23        1.71  2.43               15.6      127.0           2.80   
1      13.20        1.78  2.14               11.2      100.0           2.65   
2      13.16        2.36  2.67               18.6      101.0           2.80   
3      14.37        1.95  2.50               16.8      113.0           3.85   
4      13.24        2.59  2.87               21.0      118.0           2.80   
..       ...         ...   ...                ...        ...            ...   
173    13.71        5.65  2.45               20.5       95.0           1.68   
174    13.40        3.91  2.48               23.0      102.0           1.80   
175    13.27        4.28  2.26               20.0      120.0           1.59   
176    13.17        2.59  2.37               20.0      120.0           1.65   
177    14.13        4.10  2.74               24.5       96.0           2.05   

     flavanoids  nonflavanoid_phenols  proanthocyanins  color_intensity   hue  \
0          3.06                  0.28             2.29             5.64  1.04   
1          2.76                  0.26             1.28             4.38  1.05   
2          3.24                  0.30             2.81             5.68  1.03   
3          3.49                  0.24             2.18             7.80  0.86   
4          2.69                  0.39             1.82             4.32  1.04   
..          ...                   ...              ...              ...   ...   
173        0.61                  0.52             1.06             7.70  0.64   
174        0.75                  0.43             1.41             7.30  0.70   
175        0.69                  0.43             1.35            10.20  0.59   
176        0.68                  0.53             1.46             9.30  0.60   
177        0.76                  0.56             1.35             9.20  0.61   

     od280/od315_of_diluted_wines  proline  
0                            3.92   1065.0  
1                            3.40   1050.0  
2                            3.17   1185.0  
3                            3.45   1480.0  
4                            2.93    735.0  
..                            ...      ...  
173                          1.74    740.0  
174                          1.56    750.0  
175                          1.56    835.0  
176                          1.62    840.0  
177                          1.60    560.0  

[178 rows x 13 columns]

In [6]:
df['target'] = wine.target
df

alcohol  malic_acid   ash  alcalinity_of_ash  magnesium  total_phenols  \
0      14.23        1.71  2.43               15.6      127.0           2.80   
1      13.20        1.78  2.14               11.2      100.0           2.65   
2      13.16        2.36  2.67               18.6      101.0           2.80   
3      14.37        1.95  2.50               16.8      113.0           3.85   
4      13.24        2.59  2.87               21.0      118.0           2.80   
..       ...         ...   ...                ...        ...            ...   
173    13.71        5.65  2.45               20.5       95.0           1.68   
174    13.40        3.91  2.48               23.0      102.0           1.80   
175    13.27        4.28  2.26               20.0      120.0           1.59   
176    13.17        2.59  2.37               20.0      120.0           1.65   
177    14.13        4.10  2.74               24.5       96.0           2.05   

     flavanoids  nonflavanoid_phenols  proanthocyanins  color_intensity   hue  \
0          3.06                  0.28             2.29             5.64  1.04   
1          2.76                  0.26             1.28             4.38  1.05   
2          3.24                  0.30             2.81             5.68  1.03   
3          3.49                  0.24             2.18             7.80  0.86   
4          2.69                  0.39             1.82             4.32  1.04   
..          ...                   ...              ...              ...   ...   
173        0.61                  0.52             1.06             7.70  0.64   
174        0.75                  0.43             1.41             7.30  0.70   
175        0.69                  0.43             1.35            10.20  0.59   
176        0.68                  0.53             1.46             9.30  0.60   
177        0.76                  0.56             1.35             9.20  0.61   

     od280/od315_of_diluted_wines  proline  target  
0                            3.92   1065.0       0  
1                            3.40   1050.0       0  
2                            3.17   1185.0       0  
3                            3.45   1480.0       0  
4                            2.93    735.0       0  
..                            ...      ...     ...  
173                          1.74    740.0       2  
174                          1.56    750.0       2  
175                          1.56    835.0       2  
176                          1.62    840.0       2  
177                          1.60    560.0       2  

[178 rows x 14 columns]

In [7]:
df[50:70]

alcohol  malic_acid   ash  alcalinity_of_ash  magnesium  total_phenols  \
50    13.05        1.73  2.04               12.4       92.0           2.72   
51    13.83        1.65  2.60               17.2       94.0           2.45   
52    13.82        1.75  2.42               14.0      111.0           3.88   
53    13.77        1.90  2.68               17.1      115.0           3.00   
54    13.74        1.67  2.25               16.4      118.0           2.60   
55    13.56        1.73  2.46               20.5      116.0           2.96   
56    14.22        1.70  2.30               16.3      118.0           3.20   
57    13.29        1.97  2.68               16.8      102.0           3.00   
58    13.72        1.43  2.50               16.7      108.0           3.40   
59    12.37        0.94  1.36               10.6       88.0           1.98   
60    12.33        1.10  2.28               16.0      101.0           2.05   
61    12.64        1.36  2.02               16.8      100.0           2.02   
62    13.67        1.25  1.92               18.0       94.0           2.10   
63    12.37        1.13  2.16               19.0       87.0           3.50   
64    12.17        1.45  2.53               19.0      104.0           1.89   
65    12.37        1.21  2.56               18.1       98.0           2.42   
66    13.11        1.01  1.70               15.0       78.0           2.98   
67    12.37        1.17  1.92               19.6       78.0           2.11   
68    13.34        0.94  2.36               17.0      110.0           2.53   
69    12.21        1.19  1.75               16.8      151.0           1.85   

    flavanoids  nonflavanoid_phenols  proanthocyanins  color_intensity   hue  \
50        3.27                  0.17             2.91             7.20  1.12   
51        2.99                  0.22             2.29             5.60  1.24   
52        3.74                  0.32             1.87             7.05  1.01   
53        2.79                  0.39             1.68             6.30  1.13   
54        2.90                  0.21             1.62             5.85  0.92   
55        2.78                  0.20             2.45             6.25  0.98   
56        3.00                  0.26             2.03             6.38  0.94   
57        3.23                  0.31             1.66             6.00  1.07   
58        3.67                  0.19             2.04             6.80  0.89   
59        0.57                  0.28             0.42             1.95  1.05   
60        1.09                  0.63             0.41             3.27  1.25   
61        1.41                  0.53             0.62             5.75  0.98   
62        1.79                  0.32             0.73             3.80  1.23   
63        3.10                  0.19             1.87             4.45  1.22   
64        1.75                  0.45             1.03             2.95  1.45   
65        2.65                  0.37             2.08             4.60  1.19   
66        3.18                  0.26             2.28             5.30  1.12   
67        2.00                  0.27             1.04             4.68  1.12   
68        1.30                  0.55             0.42             3.17  1.02   
69        1.28                  0.14             2.50             2.85  1.28   

    od280/od315_of_diluted_wines  proline  target  
50                          2.91   1150.0       0  
51                          3.37   1265.0       0  
52                          3.26   1190.0       0  
53                          2.93   1375.0       0  
54                          3.20   1060.0       0  
55                          3.03   1120.0       0  
56                          3.31    970.0       0  
57                          2.84   1270.0       0  
58                          2.87   1285.0       0  
59                          1.82    520.0       1  
60                          1.67    680.0       1  
61                          1.59    450.0       1  
62      

In [43]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train,y_test = train_test_split(wine.data,wine.target, test_size=0.18) 

In [13]:
from sklearn.naive_bayes import GaussianNB, MultinomialNB
mg = GaussianNB()

In [17]:
mg.fit(X_train,y_train)

GaussianNB()

In [18]:
mg.score(X_test,y_test)

1.0

### When the test size of the model is 15%, the score for the GaussianNB model is 100%

In [19]:
mn = MultinomialNB()

In [44]:
mn.fit(X_train,y_train)

MultinomialNB()

In [45]:
mn.score(X_test,y_test)

0.8787878787878788

### ### When the test size of the model is 18%, the score for the multinomialNB model is 88%